doc/sphinx/language/core/modules.rst

```diff
@@ -1010,3 +1010,66 @@ subdirectories of path). See the command :cmd:`Declare ML Module` in
 
 See :ref:`command-line-options` for a more general view over the Coq command
 line options.
+
+.. _controlling-locality-of-commands:
+
+Controlling the scope of commands with locality attributes
+----------------------------------------------------------
+
+Many commands produce an effect that will persist in a given scope,
+which can typically be the one of the section or of the module in
+which the command was called. Locality :term:`attributes <attribute>`
+can be used to change the scope of this effect. Note that some
+commands, like :cmd:`Arguments` do not have the same default locality
+when they are used inside or outside sections.
+
+.. attr:: local
+
+   In most cases, the :attr:`local` attribute limits the effect of the
+   command to the current scope (section or module).
+
+   The only exception is when applied to :cmd:`Definition`,
+   :cmd:`Theorem` or variants. For these commands, the effect of the
+   :attr:`local` attribute will be that the defined object will not be
+   available through a shorter name when importing the module in which
+   it is defined.
+
+   There is an alternative syntax for the :attr:`local` attribute
+   using the ``Local`` prefix (see :n:`@legacy_attr`).
+
+   .. note::
+
+      For some commands, this is the only supported locality within
+      sections (e.g., for :cmd:`Notation`, :cmd:`Ltac` and :ref:`Hint
+      <creating_hints>` commands).
```

Comment:
"only supported locality"??  Not clear what is meant.  If you mean that Notation, etc don't allow global or export, say that more directly, and maybe under the description of those attributes

Reply:
These commands come with a locality (either the default one or the one set by the attribute). I think this explanation belongs here. It wouldn't make sense to repeat it for every other locality attribute, especially as we extend the list.

Reply:
Perhaps "locality" should be "locality option" everywhere since "locality" is not a defined term.  The dictionary definition is "the position or site of something" though in CS it's used a bit more broadly.

Reply:
The more appropriate term should really be "scope", but my goal in this PR is not to revise the entire thing and locality is the term that has been used so far. In most places, I write "locality attribute" which is a tad more explicit.